driver-sync/src/test/functional/com/mongodb/client/ChangeStreamProseTest.java

```diff
@@ -0,0 +1,203 @@
+
+        MongoCursor<ChangeStreamDocument<Document>> cursor = collection.watch(asList(Aggregates.project(Document.parse("{ _id : 0 }"))))
+                .iterator();
+        try {
+            collection.insertOne(Document.parse("{ x: 1 }"));
+            assertNull(cursor.next());
+        } catch (MongoChangeStreamException e) {
+            exceptionFound = true;
+        } finally {
+            cursor.close();
+        }
+        assumeTrue(exceptionFound);
+    }
+
+    //
+    // Test that the ChangeStream will automatically resume one time on a resumable error (including not master)
+    // with the initial pipeline and options, except for the addition/update of a resumeToken.
+    //
+    @Test
+    public void testResumeOneTimeOnError() {
+        assumeTrue(serverVersionAtLeast(4, 0));
+        setFailPoint("getMore", 10107);
+        MongoCursor<ChangeStreamDocument<Document>> cursor = collection.watch().iterator();
+        try {
+            collection.insertOne(Document.parse("{ x: 1 }"));
+            assertNotNull(cursor.next());
+        } finally {
+            cursor.close();
+        }
+    }
+
+    //
+    // Test that ChangeStream will not attempt to resume on any error encountered while executing an aggregate command.
+    //
+    @Test
+    public void testNoResumeForAggregateErrors() {
+        boolean exceptionFound = false;
+        try {
+            collection.watch(asList(Document.parse("{ $unsupportedStage: { _id : 0 } }"))).iterator();
+        } catch (MongoCommandException e) {
+            exceptionFound = true;
+        }
+        assertTrue(exceptionFound);
+    }
+
+    //
+    // ChangeStream will not attempt to resume after encountering error code 11601 (Interrupted), 136 (CappedPositionLost),
+    // or 237 (CursorKilled) while executing a getMore command.
+    //
+    @Test
+    public void testNoResumeErrors() {
+        assumeTrue(serverVersionAtLeast(4, 0));
+        MongoCursor<ChangeStreamDocument<Document>> cursor = collection.watch().iterator();
+        collection.insertOne(Document.parse("{ x: 1 }"));
+
+        for (int errCode : asList(136, 237, 11601)) {
+            try {
+                setFailPoint("getMore", errCode);
+                cursor.next();
+            } catch (MongoException e) {
+                assertEquals(errCode, e.getCode());
+            }
+        }
+        cursor.close();
+    }
+
+    //
+    // Ensure that a cursor returned from an aggregate command with a cursor id and an initial empty batch
+    // is not closed on the driver side.
+    //
+    @Test
+    public void testCursorNotClosed() {
+        MongoCursor<ChangeStreamDocument<Document>> cursor = collection.watch().iterator();
+        assertNotNull(cursor.getServerCursor());
+    }
+
+    //
+    // Test that the killCursors command sent during the "Resume Process" must not be allowed to throw an exception.
+    //
+    @Test
+    public void testNoExceptionFromKillCursors() {
```

Comment:
Does this test actually send a killCursors command?  

I'm not 100 % how we (the java driver) can actually automate the testing of this and ensure that kill cursors is sent during the resume process.

Reply:
I have removed the test. After debugging, I don't see a clear way for the Java driver to test `killCursors`.